FLEX pager app

timelf123 opened the pull request:
## Brief description what you did
Adds Flex app. Could really use some eyeballs as I took some liberties to get a working MVP (external baseband PFLX.bin must be copied from build folder to root of sd card)

After I got this MVP working, I attempted to port into a combined baseband POCSAG and FLEX, renaming the POCSAG app to Pager. That PR is #2884 

<img width="240" height="320" alt="image" src="https://github.com/user-attachments/assets/d4361fc7-936c-4a90-8576-8ea5c6a06316" />


## Checklist
- [x]  Kept changes minimal and limited to necessary files
- [x]  Verified functionality remains intact and code compiles

timelf123 committed:
Add FLEX pager support

- Introduced a new FLEX configuration function in baseband_api.
- Added FLEX application view and associated UI elements.
- Implemented FLEX processing logic in proc_flex, including demodulation and message handling.
- Updated CMakeLists to include new FLEX source files and headers.
- Enhanced message system to support FLEX-specific messages and statistics.

This commit lays the groundwork for FLEX pager functionality, allowing for the reception and processing of FLEX messages.

RocketGod-git committed:
Fixed baseband and moved app to external with some other fixes.

RocketGod-git committed:
Format code

RocketGod-git reviewed (commented):
Everything good now. 😎

RocketGod-git committed:
Rearrange UI element initialization in FlexAppView

Moved the initialization of field_frequency above other UI elements and updated comments to clarify row positions. Adjusted the Console's position and size to start at row 1 and extend to the bottom of the screen.
Dynamic screen size added with macros for PortaRF

gullradriel reviewed (approved):
Good to go !
Thanks 👍
